src/ClientData/include/ClientData/ModuleIdentifier.h

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef CLIENT_DATA_MODULE_IDENTIFIER_H_
+#define CLIENT_DATA_MODULE_IDENTIFIER_H_
+
+#include <string>
+#include <tuple>
+
+namespace orbit_client_data {
+
+struct ModuleIdentifier {
+  std::string file_path;
```

Comment:
Should this be a std::filesystem::path instead? Otherwise we could generate distinct ModuleIdentifiers by using forward or backward slashes: 

ModuleIdentifier{"C:\a\b\c", "1234"}
ModuleIdentifier{"C:\a/b/c", "1234"}

Reply:
I thought about this before and I think you are right and I want to change this, but I will do it in a later PR. Mainly because as the next step I want this to be a replacement for `std::pair<std::string, std::string>` (and `const std::string& module_path, const std::string& build_id`). And `ModuleData.file_path()` returns also `std::string`. 

I should've written something about this into the PR description.

I will add a TODO here